datajunction-server/datajunction_server/models/materialization.py

```diff
@@ -41,6 +44,34 @@
 }
 
 
+class MaterializationStrategy(str, enum.Enum):
+    """
+    Materialization strategies
+    """
+
+    # Typically used for smaller transforms tables, where for each run we
+    # fully overwrite the table.
+    # Availability state: single table
+    FULL = "full"
+
+    # Full materialization into a new snapshot for each scheduled materialization run.
+    # DJ will manage the location of the snapshot tables and return the appropriate snapshot
+    # when requested
+    # Availability state: multiple tables, one per snapshot. The snapshot key should be based on
```

Comment:
For dimension data tracking (and for other purposes) we could have a single table with a SNAPSHOT_PARTITION here. I have done both of these options (e.g. in the TableSnapshot job type) and you can also see that described in https://docs.google.com/document/d/1B5wu-1pgjvEqtG8Q9-4Nfi7DH1dJclOOXBpBvYV5pbk/edit#heading=h.qamncy4l751q under `_day_d` suffix pattern. 

With a single (partitioned) table there could always be a problem of a schema mismatch if the schema changes, but I solved it by creating a new snapshot table in such case.

Long story short... now you know, but feel free to keep it simple for now.